### PR TITLE
Todd ferris/adding2 button controlled motors

### PR DESCRIPTION
**Pull request by ToddFerris**
Added IntakeSubsystem/Command and ShooterSubsystem/Command. Currently set up for testing with dummy ports. Will need to put proper port numbers in.

### CODE REVIEW

File: simgui-ds.json [hunk 1]
```diff
@@ -0,0 +1,98 @@
+{
```
Comment (MackinnonBuck) [inline simgui-ds.json]: Is this file needed? Seems like it was generated by running the simulator. Maybe we could add this (and the other one) to the `.gitignore`.

File: simgui.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{
```
Comment (MackinnonBuck) [inline simgui.json]: Same as the other file, maybe we could add to the `.gitignore`, unless we actually want these.

File: src/main/java/frc/robot/RobotContainer.java [hunk 3]
```diff
@@ -44,6 +55,10 @@
      */
     public RobotContainer() {
         driveSubsystem.setDefaultCommand(driveFromController);
+        shooterCommand.addRequirements(shooterSubsystem);
+        intakeCommand.addRequirements(intakeSubsystem);
```
Comment (MackinnonBuck) [inline src/main/java/frc/robot/RobotContainer.java]:
```suggestion
```

I think requirements are already added in the `ShooterCommand` and `IntakeCommand` constructors.

File: src/main/java/frc/robot/commands/IntakeCommand.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
+package frc.robot.commands;
```
Comment (MackinnonBuck) [inline src/main/java/frc/robot/commands/IntakeCommand.java:1]: Existing files use 4 spaces for indentation, could we change this one to use 4 as well? Same goes for other files indented with 2 spaces.

File: src/main/java/frc/robot/commands/IntakeCommand.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+import java.util.function.DoubleSupplier;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.IntakeSubsystem;
+
+public class IntakeCommand extends CommandBase {
+  IntakeSubsystem intakeSubsystem;
+  DoubleSupplier speed;
```
Comment (MackinnonBuck) [inline src/main/java/frc/robot/commands/IntakeCommand.java]:
```suggestion
  private final IntakeSubsystem intakeSubsystem;
  private final DoubleSupplier speed;
```

File: src/main/java/frc/robot/commands/ShooterCommand.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.ShooterSubsystem;
+
+public class ShooterCommand extends CommandBase {
+  ShooterSubsystem shooterSubsystem;
```
Comment (MackinnonBuck) [inline src/main/java/frc/robot/commands/ShooterCommand.java]:
```suggestion
  private final ShooterSubsystem shooterSubsystem;
```

File: src/main/java/frc/robot/commands/ShooterCommand.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+  private static final double SHOOTER_SPEED = 0.5; //TODO Find shooter speed
```
Comment (MackinnonBuck) [inline src/main/java/frc/robot/commands/ShooterCommand.java]: Nit: The other files have constants like this at the top of the class declaration, before class members.

File: src/main/java/frc/robot/subsystems/IntakeSubsystem.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+import edu.wpi.first.wpilibj.motorcontrol.MotorControllerGroup;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Ids;
+
+public class IntakeSubsystem extends SubsystemBase {
+  private final CANSparkMax intakeLeadMotor = new CANSparkMax(Ids.INTAKE_LEAD_DEVICE, MotorType.kBrushless);
+  private final MotorControllerGroup intakeMotors = new MotorControllerGroup(intakeLeadMotor);
+
+
+  /** Creates a new Shooter. */
```
Comment (MackinnonBuck) [inline src/main/java/frc/robot/subsystems/IntakeSubsystem.java]:
```suggestion
  /** Creates a new Intake. */
```

Or:

```suggestion
/**
 * Constructs a new {@link IntakeSubsystem} instance.
 */
```

**Review by MackinnonBuck — CHANGES_REQUESTED** [review]
Just some small things.

File: src/main/java/frc/robot/RobotContainer.java (hunk 3 above)
Comment (ToddFerris) [reply src/main/java/frc/robot/RobotContainer.java]: Lines deleted, per request.

File: src/main/java/frc/robot/subsystems/IntakeSubsystem.java (hunk 8 above)
Comment (ToddFerris) [reply src/main/java/frc/robot/subsystems/IntakeSubsystem.java]: Done.

File: src/main/java/frc/robot/commands/ShooterCommand.java (hunk 7 above)
Comment (ToddFerris) [reply src/main/java/frc/robot/commands/ShooterCommand.java]: Done.

File: src/main/java/frc/robot/commands/IntakeCommand.java (hunk 4 above)
Comment (ToddFerris) [reply src/main/java/frc/robot/commands/IntakeCommand.java:1]: Done.

File: simgui.json (hunk 2 above)
Comment (ToddFerris) [reply simgui.json]: Done.

File: src/main/java/frc/robot/commands/IntakeCommand.java [hunk 9]
```diff
@@ -0,0 +1,43 @@
+package frc.robot.commands;
+
+import java.util.function.DoubleSupplier;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.IntakeSubsystem;
+
+public class IntakeCommand extends CommandBase {
+    private IntakeSubsystem intakeSubsystem;
+    private DoubleSupplier speed;
```
Comment (MackinnonBuck) [inline src/main/java/frc/robot/commands/IntakeCommand.java]:
```suggestion
    private final IntakeSubsystem intakeSubsystem;
    private final DoubleSupplier speed;
```

File: src/main/java/frc/robot/commands/ShooterCommand.java [hunk 10]
```diff
@@ -0,0 +1,42 @@
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.ShooterSubsystem;
+
+public class ShooterCommand extends CommandBase {
+
+    private static final double SHOOTER_SPEED = 0.5; // TODO Find shooter speed
+
+    ShooterSubsystem shooterSubsystem;
```
Comment (MackinnonBuck) [inline src/main/java/frc/robot/commands/ShooterCommand.java]:
```suggestion
    private final ShooterSubsystem shooterSubsystem;
```

**Review by MackinnonBuck — APPROVED** [review]
👍
